feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/overview/webViewController/interceptors/TopUpRequestInterceptor.kt

```diff
@@ -0,0 +1,128 @@
+package io.novafoundation.nova.feature_assets.presentation.novacard.overview.webViewController.interceptors
+
+import android.webkit.WebResourceRequest
+import com.google.gson.Gson
+import java.math.BigDecimal
+import okhttp3.OkHttpClient
+
+class TopUpRequestInterceptorFactory(
+    private val okHttpClient: OkHttpClient,
+    private val gson: Gson
+) {
+    fun create(callback: TopUpRequestInterceptor.Callback): TopUpRequestInterceptor {
+        return TopUpRequestInterceptor(okHttpClient, gson, callback)
+    }
+}
+
+class TopUpRequestInterceptor(
+    private val okHttpClient: OkHttpClient,
+    private val gson: Gson,
+    private val onCardCreatedListener: Callback
+) : NovaCardInterceptor {
+
+    interface Callback {
+
+        fun onTopUpCompleted(orderId: String)
+
+        fun onTopUpStart(orderId: String, amount: BigDecimal, address: String)
+    }
+
+    override fun intercept(request: WebResourceRequest): Boolean {
+        val url = request.url.toString()
+
+        val pattern = Regex("https://api\\.mercuryo\\.io/[a-zA-Z0-9.]+/widget/sell-request/([a-zA-Z0-9]+)/status.*")
```

Comment:
This is much better approach. I think we should adapt it in `CardCreationInterceptor` as well